mzLib/Omics/BioPolymerGroup/IBioPolymerGroup.cs

```diff
@@ -0,0 +1,123 @@
+        /// </summary>
+        string BioPolymerGroupName { get; }
+
+        /// <summary>
+        /// Aggregated score for the group, computed from member PSMs or peptides.
+        /// Higher scores typically indicate higher confidence.
+        /// </summary>
+        double BioPolymerGroupScore { get; set; }
+
+        /// <summary>
+        /// All biopolymer sequences with set modifications identified in this group,
+        /// including those shared with other groups.
+        /// </summary>
+        HashSet<IBioPolymerWithSetMods> AllBioPolymersWithSetMods { get; set; }
+
+        /// <summary>
+        /// Biopolymer sequences with set modifications that are unique to this group
+        /// (not shared with any other biopolymer group).
+        /// </summary>
+        HashSet<IBioPolymerWithSetMods> UniqueBioPolymersWithSetMods { get; set; }
+
+        /// <summary>
+        /// All peptide-spectrum matches (PSMs) for this group that pass the 1% FDR threshold.
+        /// </summary>
+        HashSet<ISpectralMatch> AllPsmsBelowOnePercentFDR { get; set; }
+
+        /// <summary>
+        /// The q-value (FDR-adjusted p-value) for this biopolymer group.
+        /// Lower values indicate higher confidence in the identification.
+        /// </summary>
+        double QValue { get; set; }
+
+        /// <summary>
+        /// The best (lowest) q-value among all biopolymers with set modifications in this group.
+        /// </summary>
+        double BestBioPolymerWithSetModsQValue { get; set; }
+
+        /// <summary>
+        /// The best (highest) score among all biopolymers with set modifications in this group.
+        /// </summary>
+        double BestBioPolymerWithSetModsScore { get; set; }
+
+        /// <summary>
+        /// Summary information about modifications present on members of this group.
+        /// Each string typically describes a modification type and its frequency or location.
+        /// </summary>
+        List<string> ModsInfo { get; }
+
+        /// <summary>
+        /// Dictionary mapping sample identifiers to measured intensity values for this group.
+        /// Supports both SpectraFileInfo and IsobaricQuantSampleInfo as keys.
+        /// </summary>
+        Dictionary<ISampleInfo, double> IntensitiesBySample { get; set; }
+
+        /// <summary>
+        /// All biopolymers in this group ordered alphabetically by accession.
+        /// Provides a stable, deterministic ordering for output and comparison.
+        /// </summary>
+        List<IBioPolymer> ListOfBioPolymersOrderedByAccession { get; }
+
+        /// <summary>
+        /// Returns a tab-separated header line for output files, matching the format of <see cref="ToString"/>.
+        /// </summary>
+        string GetTabSeparatedHeader();
+
+        /// <summary>
+        /// Computes or updates the <see cref="BioPolymerGroupScore"/> based on member PSMs and peptides.
+        /// </summary>
+        void Score();
```

Comment:
Can this be removed from the interface?

Reply:
In an idea world, writing would not be coupled to the object type. I'm fine leaving it for now or removing it

Reply:
In an idea world, writing would not be coupled to the object type. I'm fine leaving it for now